### ISSUE TITLE
A failing oauth stops the entire bot from launching

### ISSUE THREAD

**Issue by alazymeme** [pajbot/pajbot2#96]
```Subscriptions: []
&{false  false false   <nil> 0 false false 0 0 false 0  false  0   false    false        false false  false <nil> 0   0 false [] }
&{400 Bad Request 400 HTTP/2.0 2 0 map[Content-Type:[application/json; charset=utf-8] Date:[Wed, 05 Feb 2020 07:05:30 GMT] Server:[tsa_o] Set-Cookie:[personalization_id="v1_aP+N+LpgEvdurRtNoWPQHw=="; Max-Age=63072000; Expires=Fri, 4 Feb 2022 07:05:30 GMT; Path=/; Domain=.twitter.com guest_id=v1%3A158088633023514019; Max-Age=63072000; Expires=Fri, 4 Feb 2022 07:05:30 GMT; Path=/; Domain=.twitter.com] Strict-Transport-Security:[max-age=631138519] X-Connection-Hash:[7231bd910375320726dff1f551896777] X-Response-Time:[112]] 0xc000dae660 -1 [] false true map[] 0xc000236a00 0xc000148420}
twitter: 215 Bad Authentication data.
Error starting twitter stream: Missing twitter configuration fields
Available modules: [bad_character_filter emote_limit pajbot1_banphrase report value welcome tusecommands banned_names debug giveaway goodbye link_filter twitter punisher afk basic_commands message_length_limit other_commands message_height_limit commands bttv_emote_parser latin_filter nuke pajbot1_commands test]
Starting web on host 0.0.0.0:45334
ERROR!!! Error getting token from token source for bot psisynbot
2020/02/05 08:05:32 An error occurred while loading bots: oauth2: cannot fetch token: 400 Bad Request
Response: {"status":400,"message":"Invalid refresh token"}

**Comment by alazymeme** [comment]
I can't reproduce the above anymore, but this did happen to me the other day when I was running a banlist:

```
Sep 16 05:57:10 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got moderation action
Sep 16 06:03:15 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got moderation action
Sep 16 06:32:16 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Unparsed message: :tmi.twitch.tv HOSTTARGET #weest :sneegsnag 0
Sep 16 08:32:18 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Unparsed message: :tmi.twitch.tv HOSTTARGET #weest :- 0
Sep 16 10:30:12 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Unparsed message: :tmi.twitch.tv HOSTTARGET #alazymeme :samtwocan 0
Sep 16 10:41:50 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Connection no longer connected
Sep 16 10:41:50 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got moderation action
Sep 16 10:41:51 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got moderation action
Sep 16 10:43:22 Ubuntu-2004-focal-64-minimal start.sh[1789558]: ws handler
Sep 16 10:43:22 Ubuntu-2004-focal-64-minimal start.sh[1789558]: ws conn could not authenticate
Sep 16 10:43:22 Ubuntu-2004-focal-64-minimal start.sh[1789558]: aaaaaaaaa
Sep 16 10:43:22 Ubuntu-2004-focal-64-minimal start.sh[1789558]: new connection
Sep 16 10:43:23 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got msg: {"Type":"Subscribe","Topic":"ReportHandled"}
Sep 16 10:43:23 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got msg: {"Type":"Subscribe","Topic":"ReportReceived","Data":{"ChannelID":"130924701"}}
Sep 16 10:43:23 Ubuntu-2004-focal-64-minimal start.sh[1789558]: aaaaaaaaaaaaaaaa
Sep 16 10:43:23 Ubuntu-2004-focal-64-minimal start.sh[1789558]: no user
Sep 16 10:43:23 Ubuntu-2004-focal-64-minimal start.sh[1789558]: [ReportReceived] Failed authorization:
Sep 16 10:43:38 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Last insert ID: 1
Sep 16 10:43:39 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Error reading message: websocket: close 1001 (going away)
Sep 16 10:43:39 Ubuntu-2004-focal-64-minimal start.sh[1789558]: disconnect
Sep 16 10:43:39 Ubuntu-2004-focal-64-minimal start.sh[1789558]: ws handler
Sep 16 10:43:39 Ubuntu-2004-focal-64-minimal start.sh[1789558]: aaaaaaaaa
Sep 16 10:43:39 Ubuntu-2004-focal-64-minimal start.sh[1789558]: new connection
Sep 16 10:43:40 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got msg: {"Type":"Subscribe","Topic":"ReportHandled"}
Sep 16 10:43:40 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got msg: {"Type":"Subscribe","Topic":"ReportReceived","Data":{"ChannelID":"130924701"}}
Sep 16 10:43:40 Ubuntu-2004-focal-64-minimal start.sh[1789558]: aaaaaaaaaaaaaaaa
Sep 16 10:43:40 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Parameters: {"ChannelID":"130924701"}
Sep 16 10:43:40 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Channel ID: 130924701
Sep 16 10:43:40 Ubuntu-2004-focal-64-minimal start.sh[1789558]: User name: alazymeme
Sep 16 10:43:40 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Send reports to new connection
Sep 16 10:43:53 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got msg: {"Type":"Publish","Topic":"HandleReport","Data":{"Action":4,"ChannelID":"130924701","ReportID":2}}
Sep 16 10:43:53 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Handle report: report.handleReportMessage{Action:0x4, ChannelID:"130924701", ReportID:0x2, Duration:(*uint32)(nil), Handler:report.ReportUser{ID:>
Sep 16 10:43:53 Ubuntu-2004-focal-64-minimal start.sh[1789558]: untimeout {"Channel":"weest","Target":"felanbird"}
Sep 16 10:43:53 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Untimeout through pubsub: {Channel:weest Target:felanbird}
Sep 16 10:43:53 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Connection no longer connected
Sep 16 10:43:53 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Connection no longer connected
Sep 16 10:43:53 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Got moderation action
Sep 16 10:43:57 Ubuntu-2004-focal-64-minimal start.sh[1789558]: error: websocket: close 1006 (abnormal closure): unexpected EOF
Sep 16 10:43:57 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Error reading message: websocket: close 1006 (abnormal closure): unexpected EOF
Sep 16 10:43:57 Ubuntu-2004-focal-64-minimal start.sh[1789558]: disconnect
Sep 16 11:00:43 Ubuntu-2004-focal-64-minimal start.sh[1789558]: Last insert ID: 1
Sep 16 11:02:58 Ubuntu-2004-focal-64-minimal start.sh[1789558]: 2020/09/16 11:02:58 hehe
Sep 16 11:02:58 Ubuntu-2004-focal-64-minimal systemd[1]: pajbot2.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
Sep 16 11:02:58 Ubuntu-2004-focal-64-minimal systemd[1]: pajbot2.service: Child 1789549 belongs to pajbot2.service.
Sep 16 11:02:58 Ubuntu-2004-focal-64-minimal systemd[1]: pajbot2.service: Main process exited, code=exited, status=1/FAILURE
Sep 16 11:02:58 Ubuntu-2004-focal-64-minimal systemd[1]: pajbot2.service: Failed with result 'exit-code'.
```

**Comment by pajlada** [comment]
Should be solved by a81ec5d6f3f1928820bb89af0bd870f751806787
Make a new issue if you run into any other issues
